code/game/machinery/computer/arcade/orion.dm

```diff
@@ -0,0 +1,524 @@
+	spaceport_raided = FALSE
+
+/obj/machinery/computer/arcade/orion_trail/proc/report_player(mob/gamer)
+	if(gamers[gamer] == ORION_GAMER_GIVE_UP)
+		return // enough harassing them
+
+	if(gamers[gamer] == ORION_GAMER_PAMPHLET)
+		say("WARNING: Continued antisocial behavior detected: Dispensing self-help literature.")
+		new /obj/item/paper/pamphlet/violent_video_games(drop_location())
+		gamers[gamer]--
+		return
+
+	if(!(gamer in gamers))
+		gamers[gamer] = 0
+
+	gamers[gamer]++ // How many times the player has 'prestiged' (massacred their crew)
+
+	if(gamers[gamer] > ORION_GAMER_REPORT_THRESHOLD && prob(20 * gamers[gamer]))
+
+		radio.set_frequency(FREQ_SECURITY)
+		radio.talk_into(src, "SECURITY ALERT: Crewmember [gamer] recorded displaying antisocial tendencies in [get_area(src)]. Please watch for violent behavior.", FREQ_SECURITY)
+
+		radio.set_frequency(FREQ_MEDICAL)
+		radio.talk_into(src, "PSYCH ALERT: Crewmember [gamer] recorded displaying antisocial tendencies in [get_area(src)]. Please schedule psych evaluation.", FREQ_MEDICAL)
+
+		gamers[gamer] = ORION_GAMER_PAMPHLET //next report send a pamph
+
+		gamer.client.give_award(/datum/award/achievement/misc/gamer, gamer) // PSYCH REPORT NOTE: patient kept rambling about how they did it for an "achievement", recommend continued holding for observation
+		gamer.mind?.adjust_experience(/datum/skill/gaming, 50) // cheevos make u better
+
+		if(!isnull(GLOB.data_core.general))
+			for(var/datum/data/record/insanity_records in GLOB.data_core.general)
+				if(insanity_records.fields["name"] == gamer.name)
+					insanity_records.fields["m_stat"] = "*Unstable*"
+					return
+
+/obj/machinery/computer/arcade/orion_trail/ui_interact(mob/user, datum/tgui/ui)
+	. = ..()
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "OrionGame", name)
+		ui.open()
+
+/obj/machinery/computer/arcade/orion_trail/ui_assets(mob/user)
+	return list(
+		get_asset_datum(/datum/asset/spritesheet/moods),
+	)
+
+/obj/machinery/computer/arcade/orion_trail/ui_data(mob/user)
+	var/list/data = list()
+	data["gamestatus"] = gameStatus
+
+	data["engine"] = engine
+	data["turns"] = turns
+	data["hull"] = hull
+	data["electronics"] = electronics
+	data["food"] = food
+	data["fuel"] = fuel
+	data["lings_suspected"] = lings_suspected
+
+	data["eventname"] = event?.name
+	data["eventtext"] = event?.text
+	data["buttons"] = event?.event_responses
+
+	data["spaceport_raided"] = spaceport_raided
+
+	data["reason"] = reason
+
+	return data
+
+/obj/machinery/computer/arcade/orion_trail/ui_static_data(mob/user)
+	var/list/static_data = list()
+	static_data["gamename"] = name
+	static_data["emagged"] = obj_flags & EMAGGED
+	static_data["settlers"] = settlers
+	static_data["settlermoods"] = settlermoods
+	return static_data
+
+/obj/machinery/computer/arcade/orion_trail/ui_act(action, list/params)
+	. = ..()
+	if(.)
+		return
+	if(!iscarbon(usr))
+		return
+
+	. = TRUE
+
+	var/mob/living/carbon/gamer = usr
```

Comment:
I know you did the iscarbon check earlier but I think you should do this and then `if(!istype(gamer)) return`
instead as it means it's harder for someone to accidentally remove the iscarbon check without realising the cast exists and blowing shit the fuck up